scipy/integrate/_ivp/ivp.py

```diff
@@ -425,25 +425,25 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
             ts.append(t)
             ys.append(y)
         else:
-            t_step = []
-
-            while t_eval_i < n_eval and s * (t_eval[t_eval_i] - t) < 0:
-                t_step.append(t_eval[t_eval_i])
-                t_eval_i += 1
+            t_end = t_eval_i
+            while t_end < n_eval and s * (t_eval[t_end] - t) < 0:
+                t_end += 1
```

Comment:
I updated the code. It is note very clear, because`np.searchsorted` is quite a nuisance when you want to work with descending order. It should be correct.